tests/platforms/linux/appimage/test_build.py

```diff
@@ -299,22 +323,50 @@ def test_build_failure(build_command, first_app, tmp_path):
     )
 
     # chmod isn't invoked if the binary wasn't created.
-    assert build_command.os.chmod.call_count == 0
+    assert build_command.tools.os.chmod.call_count == 0
 
 
 @pytest.mark.skipif(
     sys.platform == "win32", reason="Windows paths aren't converted in Docker context"
 )
-def test_build_appimage_in_docker(build_command, first_app, tmp_path):
+def test_build_appimage_in_docker(build_command, first_app, tmp_path, monkeypatch):
     """A Linux app can be packaged as an AppImage in a docker container."""
+
     # Enable docker, and move to a non-Linux OS.
-    build_command.host_os = "TestOS"
+    build_command.tools.host_os = "TestOS"
     build_command.use_docker = True
 
-    build_command.build_app(first_app)
+    build_command.verify_app_tools(first_app)
 
-    # Ensure that the effect of the Docker context has been reversed.
-    assert build_command.subprocess != build_command.Docker
+    # Assert docker image built
+    build_command._subprocess.Popen.assert_any_call(
+        [
+            "docker",
+            "build",
+            "--progress",
+            "plain",
+            "--tag",
+            f"briefcase/com.example.first-app:py3.{sys.version_info.minor}",
+            "--file",
+            f"{tmp_path / 'base_path' / 'linux' / 'appimage' / 'First App' / 'Dockerfile'}",
+            "--build-arg",
+            f"PY_VERSION=3.{sys.version_info.minor}",
+            "--build-arg",
+            "SYSTEM_REQUIRES=",
+            "--build-arg",
+            "HOST_UID=1000",
+            "--build-arg",
+            "HOST_GID=1001",
+            f"{tmp_path / 'base_path' / 'src'}",
+        ],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        bufsize=1,
+        text=True,
+        encoding=mock.ANY,
+    )
```

Comment:
This definitely needs to be tested; but it should be a separate test of verifying the app tools. This test of `verify_app_tools()` is completely independent of the test for `build_app()`.

Reply:
Yeah....wasn't especially proud of these appimage platform test updates. They were definitely on the "refinement" list.

I removed all the tool verification setup and just provided a `DockerAppContext` directly.